app/support/order_row_importer.rb

```diff
@@ -70,10 +70,14 @@ def order_key
     @order_key ||= [user_field, chart_string_field, order_date_field]
   end
 
-  def row_with_errors # TODO: refactor
-    new_row = @row.dup
+  def row_with_errors
+    # Start with a hash of HEADERS keys with nil values to ensure optional columns
+    # are included in the report even if they are not in the uploaded CSV.
```

Comment:
Thanks. I think those are both clearer. I think I like the `each_with_object` the best. 

The `transform_values` method seems pretty nice for some cases, though (looked it up, it appears to be a Rails 4.2 extension and then is added to ruby in 2.4).